From ebb1c322b40d96012f03e65937f9c40a6a38c34f Mon Sep 17 00:00:00 2001
From: Nicola Mometto <brobronsa@gmail.com>
Date: Tue, 11 Feb 2014 18:33:18 +0100
Subject: [PATCH] Fix MAX-QUEUE-SIZE type hint: def evaluates the metadata on
 the symbol so the type hint needs to be quoted

---
 src/main/clojure/clojure/core/async/impl/protocols.clj | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/clojure/clojure/core/async/impl/protocols.clj b/src/main/clojure/clojure/core/async/impl/protocols.clj
index 2b30117..4d31309 100644
--- a/src/main/clojure/clojure/core/async/impl/protocols.clj
+++ b/src/main/clojure/clojure/core/async/impl/protocols.clj
@@ -10,7 +10,7 @@
   clojure.core.async.impl.protocols)
 
 
-(def ^:const ^int MAX-QUEUE-SIZE 1024)
+(def ^:const ^{:tag 'int} MAX-QUEUE-SIZE 1024)
 
 (defprotocol ReadPort
   (take! [port fn1-handler] "derefable val if taken, nil if take was enqueued"))
-- 
1.8.5.4

